scripts/polyrith_sage_helper.py

```diff
@@ -15,20 +17,41 @@ def power_to_string(var: int, pow: int) -> str:
         return var_s
     return mk_app("poly.pow", var_s, str(pow))
 
-def sum_to_string(terms: list[str]) -> str:
-    terms = iter(terms)
+@dataclass
+class MonomForm:
+    pos_form: str
+    neg_form: Optional[str] = None
+
+def sum_to_string_aux(old: str, nxt: MonomForm) -> str:
+    return mk_app("poly.sub" if nxt.neg_form is not None else "poly.add", old, nxt.pos_form)
+
+def sum_to_string(terms: Iterator[MonomForm]) -> str:
     try:
         first = next(terms)
     except StopIteration:
         return const_to_string(QQ(0))
     else:
-        return reduce(lambda s, t: mk_app("poly.add", s, t), terms, first)
+        first_form = first.neg_form if first.neg_form is not None else first.pos_form
+        return reduce(sum_to_string_aux, terms, first_form)
```

Comment:
Probably clearer to inline the `aux` definition above
```suggestion
        return reduce(
            lambda old, nxt: mk_app("poly.sub" if nxt.neg_form is not None else "poly.add", old, nxt.pos_form),
            terms, first_form)
```